go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go

```diff
@@ -2370,4 +2380,322 @@ var MultiDbTransactionTests = []queries.ScriptTest{
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select active_branch()",
+				Expected: []sql.Row{{"main"}},
+			},
+			{
+				Query:    "commit",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from t1 order by a",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:            "call dolt_checkout('b1')",
+				SkipResultsCheck: true,
+			},
+			{
+				Query: "select * from t1 order by a",
+				Expected: []sql.Row{
+					{1}, {2},
+				},
+			},
+		},
+	},
+	{
+		Name: "committing to another database",
+		SetUpScript: []string{
+			"create table t1 (a int)",
+			"call dolt_add('.')",
+			"call dolt_commit('-am', 'new table')",
+			"create database db1",
+			"use db1",
+			"create table t1 (a int)",
+			"use mydb",
+			"set autocommit = 0",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "insert into db1.t1 values (1)",
+				Expected: []sql.Row{
+					{types.OkResult{RowsAffected: 1}},
+				},
+			},
+			{
+				Query: "insert into db1.t1 values (2)",
+				Expected: []sql.Row{
+					{types.OkResult{RowsAffected: 1}},
+				},
+			},
+			{
+				Query:    "select * from t1 order by a",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from db1.t1 order by a",
+				Expected: []sql.Row{{1}, {2}},
+			},
+			{
+				Query:    "commit",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from t1 order by a",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from db1.t1 order by a",
+				Expected: []sql.Row{{1}, {2}},
+			},
+		},
+	},
+	{
+		Name: "committing to another branch on another database",
+		SetUpScript: []string{
+			"create table t1 (a int)",
+			"call dolt_add('.')",
+			"call dolt_commit('-am', 'new table')",
+			"create database db1",
+			"use db1",
+			"create table t1 (a int)",
+			"call dolt_add('.')",
+			"call dolt_commit('-am', 'new table')",
+			"call dolt_branch('b1')",
+			"use mydb",
+			"set autocommit = 0",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "insert into `db1/b1`.t1 values (1)",
+				Expected: []sql.Row{
+					{types.OkResult{RowsAffected: 1}},
+				},
+			},
+			{
+				Query: "insert into `db1/b1`.t1 values (2)",
+				Expected: []sql.Row{
+					{types.OkResult{RowsAffected: 1}},
+				},
+			},
+			{
+				Query:    "select * from t1 order by a",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from db1.t1 order by a",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from `db1/b1`.t1 order by a",
+				Expected: []sql.Row{{1}, {2}},
+			},
+			{
+				Query:    "commit",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from t1 order by a",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from db1.t1 order by a",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "select * from `db1/b1`.t1 order by a",
+				Expected: []sql.Row{{1}, {2}},
+			},
+		},
+	},
+	{
+		Name: "committing to more than one branch at a time",
```

Comment:
It seems like a sharp edge to allow people to modify multiple branches but not let them SQL commit to multiple branches. If someone got into this situation without knowing the rules, it might be hard for them to unwind their DB and undo the change, unless they checked every branch. Might be painful for customers with many many branches. 

This is the cross branch commit you mentioned as a fast-follow? (I think I originally got confused in the PR description between Dolt commits and SQL commits)